src/Bicep.Core/Semantics/NameBindingVisitor.cs

```diff
@@ -369,7 +369,7 @@ public override void VisitForSyntax(ForSyntax syntax)
                 if (scope.ScopeResolution == ScopeResolution.InheritFunctionsOnly)
                 {
                     // FIXME: How can we make sure only wildcard import instance functions are included in the local scope?
-                    symbolFilter = symbol => symbol is DeclaredFunctionSymbol or ImportedFunctionSymbol or WildcardImportSymbol;
+                    symbolFilter = symbol => symbol is VariableSymbol or DeclaredFunctionSymbol or ImportedFunctionSymbol or WildcardImportSymbol;
```

Comment:
I think this is going to need to be a bit more complicated. Bicep allows some variable expressions that would be disallowed in ARM but then "inlines" them into wherever they are used. A UDF wouldn't be able to refer to one of these variables since it won't exist in ARM, and anything that would cause a variable to need to be inlined would result in an expression that isn't valid in a UDF.

Alternatively, it might be easier to raise a diagnostic from `EmitLimitationCalculator` if a variable that is referred to in a UDF requires inlining, as I think this passage may occur too early in the compilation process to be able to check if a variable symbol requires inlining.

Reply:
Thanks, added unit test for this case. Should be caught by BCP341.